internal/cloudproviders/azure/network.go

```diff
@@ -145,7 +145,6 @@ func (obj *AzureProvider) DelNetwork(storage resources.StorageFactory) error {
 
 	if len(mainStateDocument.CloudInfra.Azure.ResourceGroupName) == 0 {
 		log.Print("skipped already deleted the resource group")
-		return nil
 	} else {
 		if obj.haCluster {
```

Comment:
you can simplify it
as once the resouece group is == "" there is no point in further execution you can simply return nil

Reply:
by just removing the else and simplify the code